P/PETSc/build_tarballs.jl

```diff
@@ -33,32 +34,88 @@ build_petsc()
+        USE_SUITESPARSE=0
+    elif [[ "${1}" == "single" ]] && [[ "${2}" == "complex" ]]; then
+        USE_SUITESPARSE=0
+    fi
+
+    if [[ "${target}" == *-apple* ]]; then
+        # This fails because of a linker problem (-soname should be -dynamic_name)
+        USE_SUPERLU_DIST=0
+    fi;
+
     mkdir $libdir/petsc/${1}_${2}_${3}
+
+    if [[ "${target}" == *-mingw* ]]; then
+        # Windows machines
+        #  Note: the reason to split this up in two configures, rather than one with $FLAGS is that passing
+        #  such a $FLAG to, for example, CFLAGS=$FLAGS does not seem to work (not sure why).
+
+        USE_SUPERLU_DIST=0
+        CFLAGS="-fno-stack-protector -Wl,--enable-stdcall-fixup"
+        CXXFLAGS="-fno-stack-protector -Wl,--enable-stdcall-fixup"
+        EXTRA_ARGS=(
+            --with-superlu_dist=${USE_SUPERLU_DIST}
+            --download-superlu_dist-commit=v7.1.1
+        )
+    else
+        # Non windows
+
+        # Superlu_dist expects this:
+        CC=mpicc
+        FC=mpif90
+        CXX=mpicxx
+        CFLAGS="-fno-stack-protector"
+        CXXFLAGS="-fno-stack-protector"
+        EXTRA_ARGS=(
+            --download-mumps=${USE_MUMPS}
+            --with-scalapack=${USE_MUMPS}
+            --with-scalapack-lib=${libdir}/libscalapack32.${dlext}
+            --download-suitesparse=${USE_SUITESPARSE}
+            --with-suitesparse=${USE_SUITESPARSE}
+            --download-superlu_dist=${USE_SUPERLU_DIST}
+            --with-sowing=0
+            --with-clean=1
+        )
+    fi
+
     ./configure --prefix=${libdir}/petsc/${1}_${2}_${3} \
-        CC=${CC} \
-        FC=${FC} \
-        CXX=${CXX} \
-        COPTFLAGS='-O3' \
-        CXXOPTFLAGS='-O3' \
-        CFLAGS='-fno-stack-protector' \
-        LDFLAGS="-L${libdir}" \
-        FOPTFLAGS='-O3' \
-        --with-64-bit-indices=${USE_INT64} \
-        --with-debugging=0 \
-        --with-batch \
-        --with-blaslapack-lib=$BLAS_LAPACK_LIB \
-        --with-blaslapack-suffix="" \
-        --known-64-bit-blas-indices=0 \
-        --with-mpi-lib="${MPI_LIBS}" \
-        --known-mpi-int64_t=0 \
-        --with-mpi-include="${includedir}" \
-        --with-sowing=0 \
-        --with-precision=${1} \
-        --with-scalar-type=${2} \
-        --PETSC_ARCH=${target}_${1}_${2}_${3}
+    CC=${CC} \
+    FC=${FC} \
+    CXX=${CXX} \
+    COPTFLAGS='-O3' \
+    CXXOPTFLAGS='-O3' \
+    CFLAGS="${CFLAGS}" \
+    CPPFLAGS="${CXXFLAGS}" \
+    LDFLAGS="-L${libdir}" \
+    FOPTFLAGS='-O3' \
+    --with-64-bit-indices=${USE_INT64} \
+    --with-debugging=0 \
+    --with-batch \
+    --with-blaslapack-lib=$BLAS_LAPACK_LIB \
+    --with-blaslapack-suffix="" \
+    --known-64-bit-blas-indices=0 \
+    --with-mpi-lib="${MPI_LIBS}" \
+    --known-mpi-int64_t=0 \
+    --with-mpi-include="${includedir}" \
+    --with-precision=${1} \
+    --with-scalar-type=${2} \
+    --PETSC_ARCH=${target}_${1}_${2}_${3} \
+    "${EXTRA_FLAGS[@]}"
```

Comment:
I simplified the duplicate list of arguments to `configure`: lists like these are unmaintainable, hard to read, error-prone.  By having here all common arguments and then different ones in the variables let us more clearly see the differences between the two cases.  Please, triple check I didn't mess up anything, there were a lot of arguments to deal with.

Reply:
The windows tests fail. I had the same issue when doing this
```
CPPFLAGS="${CXXFLAGS}"
```
but it seemed to work when spelling out the options explicitly for windows. Beats me why.

Reply:
The windows tests fail. I had the same issue when doing this
```
CPPFLAGS="${CXXFLAGS}"
```
but it seemed to work when spelling out the options explicitly for windows. 
Beats me why.